crates/sui/src/sui_commands.rs

```diff
@@ -42,17 +49,101 @@ use sui_swarm_config::network_config_builder::ConfigBuilder;
 use sui_swarm_config::node_config_builder::FullnodeConfigBuilder;
 use sui_types::base_types::SuiAddress;
 use sui_types::crypto::{SignatureScheme, SuiKeyPair, ToFromBytes};
+use tempfile::tempdir;
+use tracing;
 use tracing::info;
 
+const DEFAULT_EPOCH_DURATION_MS: u64 = 60_000;
+
 #[allow(clippy::large_enum_variant)]
 #[derive(Parser)]
 #[clap(rename_all = "kebab-case")]
 pub enum SuiCommand {
-    /// Start sui network.
+    /// Start a local network in two modes: saving state between re-runs and not saving state
+    /// between re-runs. Please use (--help) to see the full description.
+    ///
+    /// By default, sui start will start a local network from the genesis blob that exists in
+    /// the Sui config default dir or in the config_dir that was passed. If the default directory
+    /// does not exist and the config_dir is not passed, it will generate a new default directory,
+    /// generate the genesis blob, and start the network.
+    ///
+    /// Note that if you want to start an indexer, Postgres DB is required.
     #[clap(name = "start")]
     Start {
+        /// Config directory that will be used to store network config, node db, keystore
+        /// sui genesis -f --with-faucet generates a genesis config that can be used to start this
+        /// proces. Use with caution as the `-f` flag will overwrite the existing config directory.
+        /// We can use any config dir that is generated by the sui genesis.
         #[clap(long = "network.config")]
-        config: Option<PathBuf>,
+        config_dir: Option<std::path::PathBuf>,
+
+        /// A new genesis is created each time this flag is set, and state is not persisted between
+        /// runs. Only use this flag when you want to start the network from scratch every time you
+        /// run this command.
+        #[clap(long)]
+        force_regenesis: bool,
+
+        /// Start a GraphQL server with default host and port: 127.0.0.1:8000. Note that GraphQL
+        /// requires a running indexer.
+        #[clap(long)]
+        with_graphql: bool,
+
+        /// Start a faucet with default host and port: 127.0.0.1:9123.
+        #[clap(long)]
+        with_faucet: bool,
+
+        /// Start an indexer with default host and port: 0.0.0.0:9124.
+        /// The indexer will be started in writer mode and reader mode.
+        #[clap(long)]
+        with_indexer: bool,
+
+        /// Port to start the Fullnode RPC server on. Default port is 9000.
+        #[clap(long, default_value = "9000")]
+        fullnode_rpc_port: u16,
+
+        /// Port to start the faucet on. Default port is 9123.
+        #[clap(long, default_value = "9123")]
+        faucet_port: u16,
+
+        /// Host to start the GraphQl server on. Default host is 127.0.0.1.
+        #[clap(long, default_value = "127.0.0.1")]
+        graphql_host: String,
```

Comment:
I don't really understand this parameter -- what other value is there for `graphql_host` other than localhost?

Reply:
I am not a networking expert, but I assume this would allow someone to start it on 0.0.0.0. We could remove it and then check for it in the `sui-test-validator` script to maintain backward compatibility.

Reply:
I guess my question is whether it makes a meaningful difference which host you give here (beyond it working if you provide a loopback address like `127.0.0.1` or not if you don't).

Reply:
Removed, we'll see if anyone complains about it.